include/dxc/Support/dxcapi.extval.h

```diff
@@ -0,0 +1,20 @@
+#include "dxc/Support/dxcapi.use.h"
+#include "dxc/dxcapi.h"
+
+class DxcDllExtValidationSupport : public dxc::DxcDllSupport {
+  // DxcDllExtValidationSupport manages the
+  // lifetime of dxcompiler.dll, while the member, m_DxilSupport,
+  // manages the lifetime of dxil.dll
+  dxc::DxcDllSupport DxilSupport;
+
+  std::string DxilDllPath;
+  // override DxcDllSupport's implementation of InitializeInternal,
+  // adding the environment variable value check for a path to a dxil.dll
+
+  std::string GetDxilDllPath() { return DxilDllPath; }
+  bool DxilDllFailedToLoad() {
+    return !DxilDllPath.empty() && !DxilSupport.IsEnabled();
+  }
+
+  HRESULT InitializeInternal(LPCSTR dllName, LPCSTR fnName);
```

Comment:
This should be override of a virtual base method (see other comment).  This method should also be moved under `protected`.
```suggestion
  HRESULT InitializeInternal(LPCSTR dllName, LPCSTR fnName) override;
```